src/app/[locale]/uikit/WeekView.tsx

```diff
@@ -0,0 +1,197 @@
+
+export function WeekView({
+  locale,
+  todayDate,
+}: {
+  locale: Locale;
+  todayDate: Date;
+}) {
+  const [originDate, setOriginDate] = useState(todayDate);
+
+  const changeWeek = (direction: "prev" | "next") => {
+    return () =>
+      setOriginDate((prevOrigin) => {
+        const multiplier = direction === "prev" ? -1 : 1;
+
+        return addDays(prevOrigin, 7 * multiplier);
+      });
+  };
+
+  const changeMonth = useCallback((monthCount: number) => {
+    const nextOriginDate = startOfMonth(addMonths(todayDate, monthCount));
+
+    setOriginDate(isPast(nextOriginDate) ? todayDate : nextOriginDate);
+  }, []);
+
+  const weekViewData = getWeekViewData({ locale, originDate });
+
+  return (
+    <Box>
+      <MonthSelect
+        locale={locale}
+        month={differenceInCalendarMonths(originDate, todayDate)}
+        changeMonth={changeMonth}
+        todayDate={todayDate}
+      />
+      <Grid
+        gridTemplateColumns="auto 1fr auto"
+        // TODO adjust layout to use common Grid parent for all the controls
+        gridTemplateAreas="'prev-week . next-week'"
+      >
+        <GridItem gridArea="prev-week">
+          <Button
+            type="button"
+            variant="ghost"
+            onClick={changeWeek("prev")}
+            disabled={isPast(originDate)}
+          >
+            <GrNext size={30} className={css({ rotate: "180deg" })} />
+          </Button>
+        </GridItem>
+
+        <GridItem gridArea="next-week">
+          <Button
+            type="button"
+            variant="ghost"
+            onClick={changeWeek("next")}
+            disabled={
+              differenceInCalendarMonths(addDays(originDate, 7), todayDate) > 12
+            }
+          >
+            <GrNext size={30} />
+          </Button>
+        </GridItem>
+
+        <Grid gridTemplateColumns="7" gap="10px">
+          {weekViewData.map((weekdayInfo) => {
+            return (
+              <Box
+                key={weekdayInfo.weekdayShort}
+                borderBlockEnd="4px solid token(colors.primary.disabled)"
+                paddingBlockEnd="3px"
+                aria-selected={weekdayInfo.isToday}
+                css={{
+                  textAlign: "center",
+                  _selected: {
+                    borderColor: "primary",
+                  },
+                }}
+              >
+                <time
+                  className={css({
+                    textStyle: "body.1",
+                    textTransform: "capitalize",
+                    whiteSpace: "pre",
+                  })}
+                  dateTime={lightFormat(weekdayInfo.day, "yyyy-MM-dd")}
+                  suppressHydrationWarning
+                >
+                  {weekdayInfo.weekdayShort}
+                  {"\n"}
+                  {lightFormat(weekdayInfo.day, "d")}
+                </time>
+              </Box>
+            );
+          })}
+        </Grid>
```

Comment:
А тут прикольно было бы сделать как-то так:
```suggestion
        <WithWeekViewData>
          {(weekdayInfo) => {
            return (
              <Box
                key={weekdayInfo.weekdayShort}
                borderBlockEnd="4px solid token(colors.primary.disabled)"
                paddingBlockEnd="3px"
                aria-selected={weekdayInfo.isToday}
                css={{
                  textAlign: "center",
                  _selected: {
                    borderColor: "primary",
                  },
                }}
              >
                <time
                  className={css({
                    textStyle: "body.1",
                    textTransform: "capitalize",
                    whiteSpace: "pre",
                  })}
                  dateTime={lightFormat(weekdayInfo.day, "yyyy-MM-dd")}
                  suppressHydrationWarning
                >
                  {weekdayInfo.weekdayShort}
                  {"\n"}
                  {lightFormat(weekdayInfo.day, "d")}
                </time>
              </Box>
            );
          }}
        </WithWeekViewData>
```